src/os/bluestore/BlueStore.cc

```diff
@@ -3678,7 +3684,6 @@ void BlueStore::Onode::get() {
   }
 }
 void BlueStore::Onode::put() {
-  ++put_nref;
```

Comment:
A question from the un-initiated: you are no longer guaranteeing that the nref will be positive while playing around with the entry. There is a guarantee that there are no parallel threads that might react to the nref being 0 while still manipulating the object?

Reply:
The assumption is, as with any reference counting schemes, that you have to obtain ref to manipulate object.
If nref == 1 this means that the only reference is in onode_map holding OnodeRef objects.
There is only one function that can access the node now: get_onode(). It operates under the same lock that is in Onode::put().